wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -62,8 +62,8 @@ getFullWalletHistory cWalId = do
     logTxHistory "Mempool" localHistory
 
     fullHistory <- addRecentPtxHistory cWalId $ localHistory `Map.union` blockHistory
+    walAddrsDetector <- getWalletAddrsDetector Ever cWalId
```

Comment:
It appears that we are potentially making things slower here.

Previously: we had `cAddrs` and the check was `flip S.member (Set.fromList cAddrs)`
Now: we had `cAddrs`, we discard them, and if `accIds` is not a singleton list, we reconstruct them in `getWalletAddrsDetector` again with `concatMapM` and `S.fromList`, only to have the do `flip S.member` again.

Reply:
This codepath clearly can be optimized further by not reconstructing `cAddrs`.

Reply:
For the cases of interest, `accIds` *is* a singleton list. We're dealing with a wallet with a single account.

Reply:
Even if we're primarily interested in the `length accIds == 1`, doesn't mean that we should make the other case slower.

My point is that this PR is not a strict improvement, but it could be with very small changes.